rust/Makefile.am

```diff
@@ -35,16 +35,19 @@ RUST_TARGET = --target $(host_triplet)
 endif
 
 all-local: Cargo.toml
+	mkdir -p $(abs_top_builddir)/rust/gen
 if HAVE_CYGPATH
 	cd $(abs_top_srcdir)/rust && \
 		@rustup_home@ CARGO_HOME="$(CARGO_HOME)" \
 		CARGO_TARGET_DIR="$(e_rustdir)/target" \
+    SURICATA_LUA_SYS_HEADER_DST="$(e_rustdir)/gen" \
 		$(CARGO) build $(RELEASE) \
 			--features "$(RUST_FEATURES)" $(RUST_TARGET)
 else
 	cd $(abs_top_srcdir)/rust && \
 		@rustup_home@ CARGO_HOME="$(CARGO_HOME)" \
 		CARGO_TARGET_DIR="$(abs_top_builddir)/rust/target" \
+    SURICATA_LUA_SYS_HEADER_DST="$(abs_top_builddir)/rust/gen" \
```

Comment:
Just noticed the messed up tabbing.  Make is cool with it as its a continuation line.  Could fix, or fix later with another PR.

Reply:
Not sure if it is related, but my freebsd14 runner failed during staging
```
Making all in rust
gmake[2]: Entering directory '/var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/rust'
mkdir -p /var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/rust/gen
cd /var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/../../rust && \
	 CARGO_HOME="/var/tmp/gitlab_runner/.cargo" \
	CARGO_TARGET_DIR="/var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/rust/target" \
    SURICATA_LUA_SYS_HEADER_DST="/var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/rust/gen" \
	/usr/local/bin/cargo build --release  \
		--features "ja3 ja4  " 
    Updating crates.io index
error: failed to write /var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/rust/Cargo.lock
Caused by:
  failed to open: /var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/rust/Cargo.lock
Caused by:
  Permission denied (os error 13)
gmake[2]: *** [Makefile:541: all-local] Error 101
gmake[2]: Leaving directory '/var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub/rust'
gmake[1]: *** [Makefile:504: all-recursive] Error 1
gmake[1]: Leaving directory '/var/tmp/gitlab_runner/builds/Tys3Svbf/0/inliniac/suricata-ci/suricata/suricata-8.0.0-dev/_build/sub'
gmake: *** [Makefile:718: distcheck] Error 1
Cleaning up project directory and file based variables 00:00
ERROR: Job failed: exit status 1
```

Reply:
staging w/o this PR passes this test, so it's very likely related to the changes in the PR

Reply:
Anything you can share about your FreeBSD setup? Install scripts, CI scripts? It works for me on FreeBSD 14, with the minimum of packages installed and just running "make distcheck" on the command line.  Seems very flaky with any -j parameter tho, even unmodified git master.
- Are you using a dist built elsewhere?
- Or is this from git?
-

Reply:
Nothing fancy. It builds a dist built by another builders
```
    - CXX=clang++ CC=clang ASAN_OPTIONS="abort_on_error=1" ac_cv_func_realloc_0_nonnull="yes" ac_cv_func_malloc_0_nonnull="yes" CFLAGS="-O0 -ggdb -fno-inline -fno-omit-frame-pointer" ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var --disable-shared --disable-gccmarch-native --disable-coccinelle --enable-ipfw --enable-netmap
    - gmake clean
    - PYTHONPATH="/usr/local/lib/python3.9/site-packages/" DISTCHECK_CONFIGURE_FLAGS="--disable-coccinelle --enable-ipfw --enable-netmap CXXFLAGS=\"-Wno-sign-compare -Wno-unused-private-field -Wno-unused-const-variable\"" gmake distcheck
```

Reply:
Hmm.. I think its because Cargo.toml was updated with the new lua crate version without updating the lock.. Sigh..

Reply:
ah that does ring a bell, iirc this scenario happened before. I don't know why this setup is so sensitive to it.